montecarlo/cfast_mcarlo.py

```diff
@@ -514,18 +522,43 @@ def _locate_in_the_middle(self, fire_room):
 
         return location
 
+    def _locate_randomly(self, fire_room):
+        location = {}
+        # locate fire in compartment
+        compa = self.s.query("SELECT * FROM aamks_geom WHERE name=?", (fire_room, ))[0]
+
+        # draw location (uniform across the compartment)
+        dx = uniform(0, compa['width'])
+        dy = uniform(0, compa['depth'])
+
+        x = int(compa['x0']+dx)
+        y = int(compa['y0']+dy)
+        self._fire_height = round((compa['height'] * (1-math.log10(uniform(10**0.1,10)))) / 100, 2)
+        z = int(self._fire_height * 100) + compa['z0']
+
+        location['global'] = [x,y,z]    #[cm]
+        location['local'] = [round(lc/100, 2) for lc in [dx, dy]]   #[m]
+        location['floor'] = compa['floor']
+        location['fire_id'] = f"f{compa['global_type_id']}"
+        self._fire_id = location['fire_id']
+
+        self._psql_log_variable('heigh', self._fire_height)
+
+        return location
+
     def _deterministic_fire(self):
         comp, loc = {}, {}
         fire = self.s.query("SELECT * FROM aamks_geom WHERE type_pri='FIRE'")[0]
-        room = self.s.query("SELECT floor,name,type_sec,global_type_id,x0,y0 FROM aamks_geom WHERE floor=? AND type_pri='COMPA' AND fire_model_ignore!=1 AND x0<=? AND y0<=? AND x1>=? AND y1>=?", 
+        room = self.s.query("SELECT floor,name,type_sec,global_type_id,x0,y0,z0 FROM aamks_geom WHERE floor=? AND type_pri='COMPA' AND fire_model_ignore!=1 AND x0<=? AND y0<=? AND x1>=? AND y1>=?", 
                 (fire['floor'], fire['x0'], fire['y0'], fire['x1'], fire['y1']))[0]
 
-        loc['global'] =  [fire['center_x'], fire['center_y'], fire['z1'] - fire['z0']]
+        loc['global'] =  [fire['center_x'], fire['center_y'], fire['z0']]
```

Comment:
why you change this? fire row has 2 columns in aamks.sqlite aamks_geom table z0 and z1. fire['z1']-fire['z0'] gives us real height of fire, only fire['z0'] will give us the Z coordinate of the bottom part of the fire